app/src/main/java/com/kickstarter/ui/activities/SurveyResponseActivity.java

```diff
@@ -0,0 +1,131 @@
+import android.support.annotation.Nullable;
+import android.support.v7.app.AlertDialog;
+import android.util.Pair;
+import android.view.View;
+import android.webkit.WebView;
+
+import com.kickstarter.R;
+import com.kickstarter.libs.BaseActivity;
+import com.kickstarter.libs.RefTag;
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel;
+import com.kickstarter.libs.utils.AnimationUtils;
+import com.kickstarter.models.Project;
+import com.kickstarter.services.KSUri;
+import com.kickstarter.services.KSWebViewClient;
+import com.kickstarter.services.RequestHandler;
+import com.kickstarter.ui.IntentKey;
+import com.kickstarter.ui.views.KSWebView;
+import com.kickstarter.viewmodels.SurveyResponseViewModel;
+
+import java.util.Arrays;
+
+import butterknife.Bind;
+import butterknife.BindString;
+import butterknife.ButterKnife;
+import okhttp3.Request;
+
+import static com.kickstarter.libs.rx.transformers.Transformers.observeForUI;
+
+@RequiresActivityViewModel(SurveyResponseViewModel.ViewModel.class)
+public class SurveyResponseActivity extends BaseActivity<SurveyResponseViewModel.ViewModel> implements KSWebViewClient.Delegate {
+  private AlertDialog confirmationDialog;
+
+  protected @Bind(R.id.survey_response_web_view) KSWebView ksWebView;
+  protected @Bind(R.id.survey_response_loading_indicator_view) View loadingIndicatorView;
+
+  protected @BindString(R.string.general_alert_buttons_ok) String okString;
+  protected @BindString(R.string.Got_it_your_survey_response_has_been_submitted) String surveyResponseSubmittedString;
+
+  @Override
+  protected void onCreate(final @Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.survey_response_layout);
+    ButterKnife.bind(this);
+
+    this.ksWebView.client().setDelegate(this);
+    this.ksWebView.client().registerRequestHandlers(
+      Arrays.asList(
+        new RequestHandler(KSUri::isProjectSurveyUri, this::handleProjectSurveyUriRequest),
+        new RequestHandler(KSUri::isProjectUri, this::handleProjectUriRequest)
+      )
+    );
+
+    this.viewModel.outputs.goBack()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(__ -> back());
+
+    this.viewModel.outputs.showConfirmationDialog()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(__ -> lazyConfirmationDialog().show());
+
+    this.viewModel.outputs.startProjectActivity()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::startProjectActivity);
+  }
+
+  private boolean handleProjectUriRequest(final @NonNull Request request, final @NonNull WebView webView) {
+    this.viewModel.inputs.projectUriRequest(request);
+    return true;
+  }
+
+  private boolean handleProjectSurveyUriRequest(final @NonNull Request request, final @NonNull WebView webView) {
+    this.viewModel.inputs.projectSurveyUriRequest(request);
+    return false;
```

Comment:
doesn't need to be addressed here but these two booleans along with the ones in `KSWebViewClient.handle_request` are still confusing to me.

Reply:
returning `false` here tells the web view client that we haven't handled the survey uri request yet, which allows for re-handling the request e.g. reloading the url on an unsuccessful form submission...but yeah it is kinda confusing we should talk irl and debate intelligently about it on our vision quest